src/Symfony/Component/Form/AbstractTypeExtension.php

```diff
@@ -45,4 +46,13 @@ public function finishView(FormView $view, FormInterface $form, array $options)
     public function configureOptions(OptionsResolver $resolver)
     {
     }
+
+    public function getExtendedType()
+    {
+        if ($this instanceof SelfConfigurableFormTypeExtension) {
+            return $this::extendsType();
+        }
+
+        throw new LogicException(sprintf('You need to implement the extendsType() method in %s.', get_class($this)));
```

Comment:
This isn't a BC break? shouldn't show a deprecation message instead?

Reply:
Throwing the exception here isn't an issue as all implementations that already exist had to implement it because of the `FormTypeExtensionInterface`. But you are right that the message is not accurate anymore.